File: src/main/java/me/nylestroke/nylemod/effect/ModEffects.java
package me.nylestroke.nylemod.effect;

import me.nylestroke.nylemod.NylemodExample;
import net.minecraft.entity.effect.StatusEffect;
import net.minecraft.entity.effect.StatusEffectCategory;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;

public class ModEffects {

    public static StatusEffect FREEZE;

    public static StatusEffect registerStatusEffect(String name) {
        return Registry.register(Registry.STATUS_EFFECT, new Identifier(NylemodExample.MOD_ID, name),
                new FreezeEffect(StatusEffectCategory.HARMFUL, 3124687));
    }

    public static void registerEffects() {
        FREEZE = registerStatusEffect("freeze");
    }

}

File: src/main/kotlin/me/nylestroke/nylemod/effect/ModEffects.kt
package me.nylestroke.nylemod.effect

import me.nylestroke.nylemod.NylemodExample
import net.minecraft.entity.effect.StatusEffect
import net.minecraft.entity.effect.StatusEffectCategory
import net.minecraft.util.Identifier
import net.minecraft.util.registry.Registry

object ModEffects {
    var FREEZE: StatusEffect? = null
    private fun registerStatusEffect(name: String?): StatusEffect {
        return Registry.register<StatusEffect, FreezeEffect>(Registry.STATUS_EFFECT, Identifier(NylemodExample.MOD_ID, name),
                FreezeEffect(StatusEffectCategory.HARMFUL, 3124687))
    }

    fun registerEffects() {
        FREEZE = registerStatusEffect("freeze")
    }
}